### PR TITLE
net: bt: Add multi-link support

### PR DESCRIPTION
**Pull request by Vudentz**
This adds support for having multiple connections based on BT_MAX_CONN.

### CODE REVIEW

**Comment by Vudentz** [comment]
@jukkar I still don't get the expected efect for default BT_MAX_CONN if NET_L2_BT

**Comment by jukkar** [comment]
I just tried this and after setting `CONFIG_BT_MAX_CONN=2` in config file, the `CONFIG_NET_IF_MAX_IPV6_COUNT` was set to 2 as expected. Or were you describing some other issue?

File: subsys/net/l2/bluetooth/bluetooth.c [hunk 1]
```diff
@@ -379,7 +427,7 @@ static bool eir_found(u8_t type, const u8_t *data, u8_t data_len,
 
 		/* TODO: Notify device address found */
 		net_mgmt_event_notify(NET_EVENT_BT_SCAN_RESULT,
-				      bt_context_data.iface);
+				      bt_context_data.conns[0].iface);
```
Comment (jukkar) [inline subsys/net/l2/bluetooth/bluetooth.c]: Is this index 0 correct here?

Comment (Vudentz) [reply subsys/net/l2/bluetooth/bluetooth.c]: Well it is only one that garanteed to exist afaik, though perhaps we tracking of which interface the scan command come from but at least the current shell will use the first interface it finds so at least it is correct in that regard.

**Comment by Vudentz** [comment]
> I just tried this and after setting `CONFIG_BT_MAX_CONN=2` in config file, the `CONFIG_NET_IF_MAX_IPV6_COUNT` was set to 2 as expected. Or were you describing some other issue?

Interesting, did you use menuconfig to change that or you had changed directly in the prj.conf? I recall trying both and not did make any difference, or perhaps this has something to do with the sample?

**Comment by jukkar** [comment]
> did you use menuconfig to change that or you had changed directly in the prj.conf?

I did it like this
```
$ cd samples/net/sockets/echo_server
$ cp overlay-bt.conf overlay-bt-test.conf
$ echo CONFIG_BT_MAX_CONN=2 >> overlay-bt-test.conf
$ mkdir test
$ cmake -Btest -DBOARD=qemu_x86 -DOVERLAY_CONFIG=overlay-bt-test.conf .
```

**Review by jukkar — APPROVED** [review]
I do not have BT setup ready right now so did not try it, but looks good.

**Pull request merged**